final-project/scripts/blog.js

```diff
@@ -0,0 +1,106 @@
+import "../styles/main.scss";
+import { URL, DOT_COLORED, DOT_UNCOLORED} from "./constants";
+
+
+class Blog {
+  constructor() {
+    this.articleNumber = 0;
+    this.prevArticle = 0;
+    this.blogArticle = null;
+    this.blogArticles = null;
+    this.url = URL;
+    this.dotColored = DOT_COLORED;
+    this.dotUncolored = DOT_UNCOLORED;
+    this.getSiteElements = new Promise(
+      (resolve) => {
+        window.onload = function () {
+          const article = {
+            title: document.getElementById("blog_title"),
```

Comment:
Just as a recommendation, when selecting nodes with js is better to do it with selectors that are not related with styles and use a prefix, ie: .js-my-class-selector